src/Juvix/Compiler/Core/Extra/Recursors/Monadic.hs

```diff
@@ -8,7 +8,7 @@ import Juvix.Compiler.Core.Extra.Recursors.Base
 umapG ::
   forall c m.
   Monad m =>
-  Collector (Int, Maybe [BinderInfo]) c ->
+  Collector (Int, [Info]) c ->
```

Comment:
why do we have `Info` instead of `BinderInfo`?

Reply:
Because now the binder's info is stored in another Info. I thought about making a type alias e.g. `type BinderInfo = Info`, but BinderInfo is already used as the wrapper and I didn't come up with a good name. Or one can return the wrapper, but then one needs to unwrap it every time.

Reply:
What's the purpose/meaning of the wrapper `BinderInfo`?

Reply:
It's technical. You can't just store something in an Info without wrapping it. It has to have a unique type whose representation is used as a key.

Reply:
I think now it's nice because I got rid of Maybe. If there's no BinderInfo then we'll just get Info.empty